### ISSUE TITLE
Scam ads

### ISSUE THREAD

**Issue by Porthos72** [jedi-gg/swgoh.gg.public#475]
Today I went on swgoh.gg to look up some stuff. But every time I clicked on something, one of those scam ads popped up in a new tab. Overall, the site has also been glitchy, logging me out, and scrolling badly as of lately

**Comment by AdmiralP** [comment]
> I'm not talking about the ads on your website those are fine. It takes me to a different tab. I would investigate it, you guys may have gotten hacked

The more likely scenario would be that you have something malicious on your system. What steps have you taken to troubleshoot it?

Don't get me wrong, it's not impossible that their site may have been hacked, but it would likely have generated a lot more users complaints. Not to mention the fact that hackers don't generally do that kind of attack on a site. They are more likely to run a DDoS attack than anything. If they were hacking to try and get at the users of a site they would generally do so in a manner that would be masked in an effort to try and get something from the end-user (passwords or other sensitive data). Again, it's not impossible. It's just unlikely.

**Comment by Porthos72** [comment]
I suppose it could be me, but it only happens on your website. I'm not
exactly sure what is going on.

**Comment by danpaulson** [comment]
@Porthos72 @AdmiralP is correct - in an age of version controlled code, along with the fact that code for our site is compiled, it is extremely unlikely we would have any access to our server without our consent (access is locked down by IP.)

Bad ads, unfortunately, are a very real issue and one that is a constant battle. Like Sanktanglia said, we've let our provider know, but generally these kinds of targeted ads come through Google, and Google takes care of them much faster than we have the ability to.

**Comment by Porthos72** [comment]
Ok, thx

**Comment by tdm00** [comment]
My intrusion protection system has alerted me to some ads on SWGOH.GG as well. I've narrowed those alerts down to this ad for Saddleback Leather Co 
![](https://cl.ly/20ecba9ebbd3/Image%2525202018-12-25%252520at%25252012.44.56%252520PM.png)

**Comment by sanktanglia** [comment]
@tdm00 ill pass it along to our ad team, thx

**Comment by Porthos72** [comment]
Ok, so this thing had mostly stopped for me. That is, until today. When attempting to look at grand Moffat tarkin's stats, a pop up would come up in the site,  and I would not be able to go back to swgoh.gg 
Here is a picture-
![screenshot_20181231-222944](https://user-images.githubusercontent.com/42918155/50570313-9c72fa80-0d4b-11e9-933b-8378eab8cd81.png)
Also,  your site is the only one I am having this happen with, so it is not me. Any ideas?

**Comment by tdm00** [comment]
This is just getting worse. I realize that the SWGOH maintainers do not have control of the ads that are displayed on their website, but that they have a need to use ads to generate income to support the costs associated with a great tool. However at this point I have to wonder if it's time to look for a different advertising source, which is in your control, rather than to knowingly allow visitors to SWGOH to be exposed to potentially malicious, and certainly unwanted, ad-based content or malware.

**Comment by danpaulson** [comment]
We're acutely aware of the issues, and there's a few reasons it seems to be getting worse:

- End of December / Q1, CPMs fall through the floor. This means that nefarious advertisers get in cheap, and this affects basically anyone that uses Google as an advertising platform (see: a large portion of the internet).

- The RevOps team that runs our ad system have been on holiday, so response time for taking down nefarious domains is slower than normal.

Trust me when I say I hate these kinds of bad ads as much as anyone, especially on something I work on. That said, we have direct access to the ad team we work on now, they have already identified and banned a long list of nefarious actors, and switching because of a short rocky period would effectively be restarting our ability to target and combat these types of ads.

TLDR; We are very aware, we're working to the best of our ability, bad ads happen but we do everything in our power to prevent or remove as fast as possible.
